--- a/tccgen.c	2026-06-06 14:48:40.453944285 +0200
+++ b/tccgen.c	2026-06-06 14:48:40.494944603 +0200
@@ -6134,7 +6134,7 @@
                            Sym **cur_field, int size_only, int al)
 {
     Sym *s, *f;
-    int index, index_last, align, l, nb_elems, elem_size;
+    int index, index_last, align, l, nb_elems, elem_size, coff;
     unsigned long corig = c;
 
     elem_size = 0;
@@ -6205,8 +6205,9 @@
     }
     /* must put zero in holes (note that doing it that way
        ensures that it even works with designators) */
-    if (!size_only && c - corig > al)
-	init_putz(sec, corig + al, c - corig - al);
+    coff = c - corig;
+    if (!size_only && coff > al)
+	init_putz(sec, corig + al, coff - al);
     decl_initializer(type, sec, c, 0, size_only);
 
     /* XXX: make it more general */
@@ -6236,8 +6237,9 @@
 	}
     }
     c += nb_elems * type_size(type, &align);
-    if (c - corig > al)
-      al = c - corig;
+    coff = c - corig;
+    if (coff > al)
+      al = coff;
     return al;
 }
 
